roles/awx/files/awx-deployment.yml

```diff
@@ -0,0 +1,133 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: awx-deployment
+parameters:
+- name: NAMESPACE
+- name: AWX_WEB_IMAGE
+- name: AWX_TASK_IMAGE
+- name: AWX_ADMIN_PASSWORD
+
+objects:
+
+- kind: Secret
+  apiVersion: v1
+  metadata:
+    name: admin-credentials
+    namespace: ${NAMESPACE}
+  stringData:
+    username: admin
+    password: ${AWX_ADMIN_PASSWORD}
+
+- apiVersion: extensions/v1beta1
+  kind: Deployment
+  metadata:
+    name: awx
+    namespace: ${NAMESPACE}
+  spec:
+    replicas: 1
+    template:
+      metadata:
+        labels:
+          name: awx-web-deploy
+          service: django
+      spec:
+        containers:
+          - name: awx-web
+            image: ${AWX_WEB_IMAGE}
+            ports:
+              - containerPort: 8052
+            volumeMounts:
+              - mountPath: /etc/tower
+                name: awx-application-config
+            env:
+              - name: DATABASE_PASSWORD
+                valueFrom:
+                  secretKeyRef:
+                    name: postgresql
+                    key: database-password
+          - name: awx-celery
+            image: ${AWX_TASK_IMAGE}
+            volumeMounts:
+              - mountPath: /etc/tower
+                name: awx-application-config
+            env:
+              - name: DATABASE_USER
+                value: awx
+              - name: DATABASE_NAME
+                value: awx
+              - name: DATABASE_HOST
+                value: postgresql
+              - name: DATABASE_PORT
+                value: "5432"
+              - name: DATABASE_PASSWORD
+                valueFrom:
+                  secretKeyRef:
+                    name: postgresql
+                    key: database-password
+              - name: AWX_ADMIN_USER
+                valueFrom:
+                  secretKeyRef:
+                    name: admin-credentials
+                    key: username
+              - name: AWX_ADMIN_PASSWORD
+                valueFrom:
+                  secretKeyRef:
+                    name: admin-credentials
+                    key: password
+          - name: awx-rabbit
+            image: rabbitmq:3
+            env:
+              - name: RABBITMQ_ERLANG_COOKIE
+                value: secretb
+              - name: RABBITMQ_NODENAME
+                value: rabbitmq
+              - name: RABBITMQ_DEFAULT_USER
+                value: awx
+              - name: RABBITMQ_DEFAULT_PASS
```

Comment:
As above re: environment/secrets

Reply:
for RABBITMQ_ERLANG_COOKIE & RABBITMQ_DEFAULT_PASS it's even worse — I left them hard-wired to fixed known values! (as do upstream awx playbooks)
AFAICT rabbitmq container has no ports exposed outside the pod — does that sound sane or should I randomize those just in case?

Reply:
Even if you want to use fixed defaults, I'd make the passwords required parameters and move the defaulting logic out into the installer bits

Reply:
Randomized these 2 as well, stored in secrets.  It seems in later AWX version rabbitmq is exposed and these will become important (?)
RABBITMQ_ERLANG_COOKIE could be mounted, not sure about RABBITMQ_DEFAULT_PASS (indirectly by putting it in `rabbitmq.conf`?); not gonna bother with this until I get guidance from awx where to invest effort.